src/dodal/beamlines/i03.py

```diff
@@ -57,494 +57,367 @@
 
 set_path_provider(PandASubpathProvider())
 
+PREFIX = BeamlinePrefix(BL)
 
-def aperture_scatterguard(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> ApertureScatterguard:
+
+@device_factory()
+def aperture_scatterguard() -> ApertureScatterguard:
     """Get the i03 aperture and scatterguard device, instantiate it if it hasn't already
     been. If this is called when already instantiated in i03, it will return the existing
     object.
     """
     params = get_beamline_parameters()
-    return device_instantiation(
-        device_factory=ApertureScatterguard,
-        name="aperture_scatterguard",
-        prefix="",
-        wait=wait_for_connection,
-        fake=fake_with_ophyd_sim,
+    return ApertureScatterguard(
+        prefix=PREFIX.beamline_prefix,
         loaded_positions=load_positions_from_beamline_parameters(params),
         tolerances=AperturePosition.tolerances_from_gda_params(params),
     )
 
 
-def attenuator(
-    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
-) -> BinaryFilterAttenuator:
+@device_factory()
+def attenuator() -> BinaryFilterAttenuator:
     """Get the i03 attenuator device, instantiate it if it hasn't already been.
     If this is called when already instantiated in i03, it will return the existing object.
     """
-    return device_instantiation(
-        BinaryFilterAttenuator,
-        "attenuator",
-        "-EA-ATTN-01:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
-    )
+    return BinaryFilterAttenuator(f"{PREFIX.beamline_prefix}-EA-ATTN-01:")
 
 
-def dcm(wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False) -> DCM:
+@device_factory()
+def dcm() -> DCM:
     """Get the i03 DCM device, instantiate it if it hasn't already been.
     If this is called when already instantiated in i03, it will return the existing object.
     """
-    return device_instantiation(
-        DCM,
+    return DCM(
+        f"{PREFIX.beamline_prefix}-MO-DCM-01:",
         "dcm",
-        "-MO-DCM-01:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
     )
 
 
+@device_factory()
 @skip_device(lambda: BL == "s03")
```

Comment:
Depending on how mx-bluesky/hyperion is handling this:

```suggestion
@device_factory(skip = BL == "s03")
```

Reply:
or `skip = (BL == "s03")` if it's any clearer what is meant

Reply:
At this point, I think we are likely going to remove all this `s03` stuff soon as I think we have pretty much given up on the simulator.

Reply:
There is a separate issue to remove the s03 system tests and I have updated that to remove s03 completely

* DiamondLightSource/mx-bluesky#631